### ISSUE TITLE
Communicate to the user that a download is allocating space and it could take a while

### ISSUE THREAD

**Issue by macgills** [kiwix/kiwix-android#1996]
**Is your feature request related to a problem? Please describe.**

When a user starts a large download there is a chance that the pre-allocation of the file will take from 30mins to 1hr and they have no idea that they should wait for that to happen.

**Describe the solution you'd like**

When a user presses stop on a pending download(note: investigate the underlying state, could be added) we should show a dialog explaining that they may have to wait.

**Describe alternatives you've considered**
We could not pre-allocate which I would have concerns over.
We could also advertise the allocating state if we have a direct state mapping for it but I think users still wouldn't know that we require them to wait a while.

**Additional context**
ticket created from #1854

**Comment by kelson42** [comment]
The only thing we know here is that the download seems (is) stuck in "Pending". The problem is that we don't really know if user face a slow download or a slow preallocating. Before starting to do a lot of work here, I would like to fix only the status in the UI. Instead of having "Pending" and then "In Progress", we would have: "Pre-allocating space", "Pending" and then "In Progress".

Of course it should be possible to stop a download in that preallocatin stage, but this is handle with #1998.

**Comment by macgills** [comment]
superseded by #2002
